account_currency/models/account_move.py

```diff
@@ -0,0 +1,20 @@
+# Copyright (C) 2025 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    @api.depends("journal_id", "statement_line_id")
+    def _compute_currency_id(self):
+        result = super()._compute_currency_id()
+        company = self.env.company
+        for invoice in self.filtered(
+            lambda rec: rec.move_type in ["out_invoice", "in_invoice"]
+        ):
+            invoice.currency_id = (
+                company.default_account_currency_id or invoice.currency_id
+            )
+        return result
```

Comment:
```suggestion
    @api.depends("journal_id", "statement_line_id")
    def _compute_currency_id(self):
        invoices = self.filtered(
            lambda rec: rec.move_type in ["out_invoice", "in_invoice"] and rec.company_id.default_account_currency_id
        )
        for invoice in invoices:
            invoices.currency_id = invoice.company_id.default_account_currency_id
        
        return super(AccountMove, (self-invoices))._compute_currency_id()
```

Reply:
I think your way is better, I mean you need call super before, then apply changes:
```python
res = super()
....
return res
```